components/apimgt/org.wso2.carbon.apimgt.rest.api.commons/src/main/java/org/wso2/carbon/apimgt/rest/api/common/util/RestApiUtil.java

```diff
@@ -526,4 +549,53 @@ public static ZoneId getRequestTimeZone(String timestamp) {
         return ZonedDateTime.parse(timestamp).getZone();
     }
 
+    /**
+     * Get defined HTTP methods in the swagger definition as space separated string
+     *
+     * @param request           Request
+     * @param serviceMethodInfo Method informations for the request
+     * @return Http Methods as a space separated string
+     * @throws APIManagementException if failed to get defined http methods
+     */
+    public static String getDefinedMethodHeadersInSwaggerContent(
+            Request request, ServiceMethodInfo serviceMethodInfo) throws APIManagementException {
+        String requestURI = request.getUri().toLowerCase(Locale.ENGLISH);
+        Swagger swagger = null;
+        Method resourceMethod;
+
+        if (requestURI.contains("/publisher")) {
+            swagger = swaggerRestAPIDefinitions.get(RestApiConstants.APPType.PUBLISHER);
+        } else if (requestURI.contains("/store")) {
+            swagger = swaggerRestAPIDefinitions.get(RestApiConstants.APPType.STORE);
+        } else if (requestURI.contains("/analytics")) {
+            swagger = swaggerRestAPIDefinitions.get(RestApiConstants.APPType.ANALYTICS);
+        } else if (requestURI.contains("/admin")) {
+            swagger = swaggerRestAPIDefinitions.get(RestApiConstants.APPType.ADMIN);
+        }
```

Comment:
Shall we also have an else condition to skip this check with returning null? Otherwise, any other REST API getting an OPTIONS call is likely to break. The caller of this has to do a null check before responding with CORS headers. WDYT?

Reply:
Thanks. Fixed with https://github.com/wso2/carbon-apimgt/pull/5095/commits/a4555fc63ff0b302cd029e49bc6910318cd21125
Shall we return an empty string? "No allowed methods".